src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,513 @@
+        } catch (NoSpaceException e) {
+            System.out.println("Please use a space to separate command, task, and time (if any)");
+        } catch (NullPointerException e) {
+            System.out.println("Sorry, this is not one of the task");
+        } catch (NumberFormatException e) {
+            System.out.println("Please use the number of the task");
+        }
+    }
+
+    public static boolean createFile(String filePath) {
+        boolean canCreate = false;
+        File newFile = new File(filePath);
+        try {
+            canCreate = newFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        return canCreate;
+    }
+
+    public static boolean createFolder(String folderPath) {
+        boolean canCreate = false;
+        File newFolder = new File(folderPath);
+        if (!newFolder.exists()) {
+            canCreate = newFolder.mkdir();
+        }
+        return canCreate;
+    }
+
+    public static void initializeHardDiskSave() {
+        if (createFolder(LIST_STORAGE_FOLDER)) {
+            System.out.println("Folder created for storing task list");
+        } else {
+            System.out.println("Nice! You already have a folder for storing task list");
+        }
+
+        if (createFile(LIST_STORAGE_FILE)) {
+            System.out.println("duke.txt created inside data folder for storing task list");
+        } else {
+            System.out.println("Nice! You already have a file inside data folder for storing task list");
+        }
+        System.out.println("Reading your List......");
+    }
+
+    public static Task convertLinetoTask(String taskLine) {
+        Task task;
+        String taskType = taskLine.substring(0, 1);
+        String taskDone = taskLine.substring(4, 5);
+        int index = taskLine.indexOf(" | ", 8);
+        String description;
+        String time = "";
+
+        if (index == -1) {
+            description = taskLine.substring(8);
+        } else {
+            description = taskLine.substring(8, index);
+            time = taskLine.substring(index + 3);
+        }
+
+        switch (taskType) {
+        case "D":
+            task = new Deadline(description, time);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        case "E":
+            task = new Event(description, time);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        default:
+            task = new ToDo(description);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        }
+        return task;
+    }
+
+    public static String convertTodotoLine(ToDo task) {
+        if (task.isDone()) {
+            return ("T | √ | " + task.getDescription());
+        } else {
+            return ("T | X | " + task.getDescription());
+        }
+    }
+
+    public static String convertDeadlinetoLine(Deadline task) {
```

Comment:
Perhaps a better approach (more OOP oriented) would be to add these functions to the respective classes of Events, ToDo etc as they provide functionalities related to those classes

Reply:
Another design approach that you can consider for your Duke would be to have separate classes like 'DataManager' that manages loading and storing of data into the files and `TaskManager` for managing the tasks instead of adding everything to the Duke class